src/Tizen.NUI/src/public/BaseComponents/ImageView.cs

```diff
@@ -1218,81 +1196,116 @@ private void SetBorder(Rectangle value)
         {
-            PropertyMap imageMap = new PropertyMap();
+            // If we set ResourceUrl as empty, Unregist visual.
+            if (key == ImageVisualProperty.URL && string.IsNullOrEmpty(_resourceUrl))
+            {
+                // If previous resourceUrl was already empty, we don't need to do anything. just ignore.
+                // Unregist and dettach Process only if previous resourceUrl was not empty
+                string currentResourceUrl = "";
+                PropertyValue currentResourceUrlValue = _imagePropertyMap?.Find(ImageVisualProperty.URL);
+                if((currentResourceUrlValue?.Get(out currentResourceUrl) ?? false) && string.IsNullOrEmpty(currentResourceUrl))
+                {
+                    PropertyValue resourceUrl = new PropertyValue(_resourceUrl);
+                    PropertyMap imageMap = new PropertyMap();
+                    imageMap.Insert(ImageVisualProperty.URL, resourceUrl);
+                    PropertyValue setValue = new PropertyValue(imageMap);
+                    SetProperty(ImageView.Property.IMAGE, setValue);
 
-            PropertyValue image = Tizen.NUI.Object.GetProperty(SwigCPtr, ImageView.Property.IMAGE);
-            image?.Get(imageMap);
-            imageMap?.Merge(fromMap);
-            PropertyValue setValue = new PropertyValue(imageMap);
-            SetProperty(ImageView.Property.IMAGE, setValue);
+                    // Image visual is not exist anymore. We should ignore lazy UpdateImage
+                    _imagePropertyUpdatedFlag = false;
+                    if(_imagePropertyUpdateProcessAttachedFlag)
+                    {
+                        ProcessorController.Instance.ProcessorOnceEvent -= UpdateImage;
+                        _imagePropertyUpdateProcessAttachedFlag = false;
+                    }
+                    // Update resourceUrl as null
+                    _imagePropertyMap[ImageVisualProperty.URL] = resourceUrl;
 
-            imageMap?.Dispose();
-            image?.Dispose();
-            setValue?.Dispose();
+                    resourceUrl?.Dispose();
+                    setValue?.Dispose();
+                    imageMap?.Dispose();
+                }
+                return;
+            }
+
+            // Update image property map value as inputed value.
+            if (key != 0)
+            {
+                if (_imagePropertyMap == null)
+                {
+                    _imagePropertyMap = new PropertyMap();
+                }
+                _imagePropertyUpdatedFlag = true;
+                _imagePropertyMap[key] = value;
+
+                // Lazy update only if _resourceUrl is not empty and ProcessAttachedFlag is false.
+                if (!string.IsNullOrEmpty(_resourceUrl) && !_imagePropertyUpdateProcessAttachedFlag)
+                {
+                    _imagePropertyUpdateProcessAttachedFlag = true;
+                    ProcessorController.Instance.ProcessorOnceEvent += UpdateImage;
```

Comment:
Hello, 
I couldn't understand the benefit, so I am just asking. ^^;
for me, it looks like delaying the process of `UpdateImage`.
is this for making secure the order of creating-destroying visuals?
could you explain a little bit more?
thank you.

Reply:
Hello world.
Previous code always try to create new visual for every `UpdateImage` call.
If we set properties 5 times (ex : `ResourceUrl`, `SynchronousLoading`, `DesiredWidth`, `DesiredHeight`, and `CornerRadius`).
It create new visual for every property set call. So there will be called 5 constructor, and 4 destructor of visuals.
And more, every proerty setting time, it also try to get 'ALL' properties of visuals. So it will be called more than 5 CreateVisualMap() function, even we don't need to know other properties.

This patch try to collect properties that  we need to re-create the visuals,
and create visual at end of event thread.
This will be called 1 constructor, and no destructor of visuals.

Also, there is one more benefit. When we want to know just one value of property,
now we don't need to get 'ALL' properties of visuals.

For `CornerRadius` and `Borderline` are special case (they will change shader code.) so I make another PR about it (#3827)